### DISCUSSION TITLE
Problem with player initialization

### DISCUSSION THREAD

**Discussion by Manthhys** [Vdauphin/HeartsAndMinds#1485] [Question and answer] [answered]
Hey guys !

I'm encountering a problem when a new player joins the server another player already connected on the server will sometimes get teleported to a random position on the map (joining player last position ?) and their stuff is reset/modified (joining player last stuff ?).

Has anyone ever experienced that ? Do you know what could provoke that in the initialization process ? I'm running a pretty much vanilla H&M just changed the faction and playable slots

Thanks in advance !
Manthys

**Comment by Vdauphin** [comment]
Hello,

Read this https://forums.bohemia.net/forums/topic/165948-mp-btc-hearts-and-minds/?page=75&tab=comments#comment-3469051

Cheers

**Reply by Manthhys** [reply]
So @Vdauphin I have this as a setting in params.hpp shouldn't it prevent that type of problems ?
`class btc_p_slot_isShare { // Each slot is share between players
        title = __EVAL(format ["      %1", localize "STR_BTC_HAM_PARAM_SLOT_ISSHARE"]);
        values[]={0,1};
        texts[]={$STR_DISABLED, $STR_ENABLED};
        default = 1;
    };`

**Reply by Vdauphin** [reply]
`default = 1`
> Now all slots available on mission start screen persist between player disconnection. This means each time a player disconnects, load out, position and medical status is saved and restored the next time this slot is occupied by the same or another player.

`default = 0`
> A mission setting allows to have for each player a different save for each slot. 

It is similar to the ticket share between faction parameter

The idea is that player will join again their last position after logout, so if they logout in the middle of nowhere, they will appear again at that precise position giving a real reason to use helicopters / vehicles to return to base. Or plan a longtime operation with limited resources

**Reply by Manthhys** [reply]
Ooh ok so I got it wrong it's the reverse for what I want to do thanks for the help !

**Reply by Manthhys** [reply]
@Vdauphin Okay I lied this didnt fix the problem, setting the parameters to each player having their own save now provoke stuff interversion upon player disconnection instead of connection, I gotta find the reason for this

**Reply by TimRuskof** [reply]
Did you find something ?

**Reply by Manthhys** [reply]
> Did you find something ?

Not yet, we're looking into the code for player persistency but no luck so far

**Reply by Vdauphin** [reply]
What is the version of H&m you are currently running?

**Reply by Manthhys** [reply]
Hey @Vdauphin @PulsarNeutronStar @TimRuskof I'm reporting back that we found the source of the problem: We used to group players slots  to have already pre-formed squads in the lobby and turns out all persistency stuff was saved and actuated on the group leader role. This happened with both persistency parameters.

We removed player slots groups and we got rid of most of the teleportation issues (although the lobby is now a mess) except when a player crashes ang gets kicked out by battleye another player sometimes get teleported and gets the disconnected player stuff.

Hope it helps !

We're using the 1.22.8 version currently but we had this issue already on 1.22.1 see here:
https://github.com/Vdauphin/HeartsAndMinds/discussions/1385#discussion-4195952

**Reply by Vdauphin** [reply]
> We're using the 1.22.8 version currently but we had this issue already on 1.22.1 see here:

I don't understand, this is impossible as 1.22 don't have this feature. This feature is here since 1.23 only

**Reply by Manthhys** [reply]
@Vdauphin it was since the first release where persistency was implemented, I might be wring on the issue version

**Comment by OverlordZorn** [comment]
We have the same issue, made a report and just got pointed this way.

We tried both parameter settings, didnt help.

We simply disabled it with this.
https://github.com/PulsarNeutronStar/HeartsAndMinds/commit/1a553e2ac1a33af2ca6f780d1afb15359c66604d

**Reply by Vdauphin** [reply]
Hello,

Please, create an issue [here](https://github.com/Vdauphin/HeartsAndMinds/issues/new?assignees=&labels=bug&template=bug_report.md&title=) if you found a bug so we can fix it for everybody

> We tried both parameter settings, didnt help.

May be you mess up something, try to found the root cause of your issue and share it please

Cheers

**Reply by OverlordZorn** [reply]
I did. It got closed instantly.

https://github.com/Vdauphin/HeartsAndMinds/issues/1486

**Reply by Vdauphin** [reply]
This report doesn't look like a bug report as it doesn't follow the issue template for bug report
Also you are describing an intended behavior of `default = 1` (see https://github.com/Vdauphin/HeartsAndMinds/discussions/1485#discussioncomment-5091997)
If you want to see this feature in action, you can join the [public develoment server](https://github.com/Vdauphin/HeartsAndMinds/discussions/1098) which is using the last release

Feel free to update and open the ticket again

**Reply by OverlordZorn** [reply]
A) 
If a Random person, which is actively playing and currently in a firefight gets teleported to a random location with a different kit when a further person joins the server and loads in is the intended behavior, i prefer not to use the feature.

B) The Issue has [BUG] in the name. If the lack of template is the issue, then maybe a different was of communcating this would be beneficial.

**Reply by Manthhys** [reply]
@PulsarNeutronStar Thanks for the deactivating code suggestion, I think we will disable it as well as it has brought more headaches than gameplay improvements. So far we had to use a Onplayerrespawn.sqf interaction to able players to return to spawn for the first 30 seconds in case they disconnected in the middle of nowhere

**Reply by Vdauphin** [reply]
> Thanks for the deactivating code suggestion,

Please test the fix #1499 instead

**Reply by Manthhys** [reply]
@Vdauphin Will do, thanks !

**Comment by Vdauphin** [comment, marked as answer]
This behavior should have been solved in last release https://github.com/Vdauphin/HeartsAndMinds/releases thanks to https://github.com/Vdauphin/HeartsAndMinds/pull/1499
Have fun 🕹 !
